src/sentry/integrations/msteams/link_identity.py

```diff
@@ -33,38 +34,47 @@ class MsTeamsLinkIdentityView(BaseView):
     @transaction_start("MsTeamsLinkIdentityView")
     @never_cache
     def handle(self, request, signed_params):
-        params = unsign(signed_params)
+        try:
+            params = unsign(signed_params)
+
+            organization, integration, idp = get_identity(
+                request.user, params["organization_id"], params["integration_id"]
+            )
 
-        organization, integration, idp = get_identity(
-            request.user, params["organization_id"], params["integration_id"]
-        )
+            if request.method != "POST":
+                return render_to_response(
+                    "sentry/auth-link-identity.html",
+                    request=request,
+                    context={"organization": organization, "provider": integration.get_provider()},
+                )
+
+            defaults = {"status": IdentityStatus.VALID, "date_verified": timezone.now()}
+            try:
+                identity, created = Identity.objects.get_or_create(
+                    idp=idp,
+                    user=request.user,
+                    external_id=params["teams_user_id"],
+                    defaults=defaults,
+                )
+                if not created:
+                    identity.update(**defaults)
+            except IntegrityError:
+                Identity.reattach(idp, params["teams_user_id"], request.user, defaults)
+
+            card = build_linked_card()
+            client = MsTeamsClient(integration)
+            user_conversation_id = client.get_user_conversation_id(
+                params["teams_user_id"], params["tenant_id"]
+            )
+            client.send_card(user_conversation_id, card)
 
-        if request.method != "POST":
             return render_to_response(
-                "sentry/auth-link-identity.html",
+                "sentry/integrations/msteams-linked.html",
                 request=request,
-                context={"organization": organization, "provider": integration.get_provider()},
+                context={"team_id": params["team_id"]},
             )
-
-        defaults = {"status": IdentityStatus.VALID, "date_verified": timezone.now()}
-        try:
-            identity, created = Identity.objects.get_or_create(
-                idp=idp, user=request.user, external_id=params["teams_user_id"], defaults=defaults
+        except (SignatureExpired, BadSignature):
```

Comment:
@NisanthanNanthakumar could we move this `except` block closer to where we call `unsign`? It will be easier to understand which line generated the error that occurred that way.